benchmark/Benchmark.h

```diff
@@ -0,0 +1,171 @@
+     *  getting measured. Needed, because otherwise nobody would be able to
+     *  tell, which time corresponds to which benchmark.
+     * @param functionToMeasure The function, that represents the benchmark.
+     *  Most of the time a lambda, that calls the actual function to benchmark
+     *  with the needed parameters.
+     */
+    void addSingleMeasurment(const std::string descriptor,
+        const std::function<void()>& functionToMeasure);
+
+    // Returns a const view of all single recorded times.
+    const std::vector<RecordEntry>& getSingleMeasurments() const;
+
+    /*
+     * @brief Creates an empty group, which can
+     *  be accesed/identified using the descriptor.
+     */
+    void addGroup(const std::string descriptor);
+
+    /*
+     * @brief Measures the time needed for the execution of the given function and
+     * saves it, together with a description, as an item in a group.
+     *
+     * @param groupDescriptor The identification of the group.
+     * @param descriptor A description, of what kind of benchmark case is
+     *  getting measured. Needed, because otherwise nobody would be able to
+     *  tell, which time corresponds to which benchmark.
+     * @param functionToMeasure The function, that represents the benchmark.
+     *  Most of the time a lambda, that calls the actual function to benchmark
+     *  with the needed parameters.
+     */
+    void addToExistingGroup(const std::string& groupDescriptor,
+        const std::string descriptor,
+        const std::function<void()>& functionToMeasure);
+
+    /*
+     * @brief Returns a const view of all the groups.
+     */
+    const ad_utility::HashMap<std::string, RecordGroup>& getGroups() const;
+
+    /*
+     * @brief Creates an empty table, which can be accesed/identified using the
+     *  descriptor.
+     *
+     * @param descriptor The name/identifier of the table.
+     * @param rowNames,columnNames The names for the rows/columns.
+     */
+    void addTable(const std::string descriptor,
+        const std::vector<std::string> rowNames,
+        const std::vector<std::string> columnNames);
+
+    /*
+     * @brief Measures the time needed for the execution of the given function and
+     * saves it as an entry in the table.
+     *
+     * @param tableDescriptor The identification of the table.
+     * @param row, column Where in the tables to write the measured time.
+     * @param functionToMeasure The function, that represents the benchmark.
+     *  Most of the time a lambda, that calls the actual function to benchmark
+     *  with the needed parameters.
+     */
+    void addToExistingTable(const std::string& tableDescriptor,
+        const size_t row, const size_t column,
+        const std::function<void()>& functionToMeasure);
+
+    /*
+     * @brief Returns a const view of all the tables.
+     */
+    const ad_utility::HashMap<std::string, RecordTable>& getTables() const;
```

Comment:
Add a simple benchmark file, that uses all the formats (SingleMeasurments, Groups, Tables), so that there is a simple test for the infrastructure.

Reply:
Done in [e3afbe6](https://github.com/ad-freiburg/qlever/pull/860/commits/e3afbe6c8fdc4ace802d10d8bfeaed63e3f338c2).